mentorship/comms/views.py

```
@@ -37,8 +37,13 @@ class CallEndHandler(APIView):
 
     def post(self, request, format=None):
         data = request.data
-        api = Api.objects.last()
-        api.receive_call_ended(data)
+        call_id = data['id']
+        end_time = data['end_time']
+        if isinstance(end_time, basestring):
+            end_time = datetime.strptime(end_time[:-6], "%Y-%m-%dT%H:%M:%S.%f")
```

We should raise a suitable error if `id` or `end_time` are not given or if `end_time` can't be parsed into a datetime.